examples/knowledge_tuning/knowledge_tuning_with_reasoning_model/reasoning_sdg_data_mixing.py

```
@@ -0,0 +1,424 @@
+    # Generate knowledge QA dataset
+    knowl_train = generate_knowledge_qa_dataset(ds, keep_context_separate=keep_context_separate, keep_document_outline=keep_document_outline)
+    
+    # Apply pretraining format if needed
+    if no_pretrain:
+        knowl_train_pretrain = knowl_train
+    else:
+        knowl_train_pretrain = knowl_train.map(_conv_pretrain, fn_kwargs={"tokenizer": tokenizer}, num_proc=10)
+    
+    # Add system messages for nemotron format
+    if nemotron_format:
+        knowl_train_pretrain = knowl_train_pretrain.map(lambda x: {'messages': [{'content': f'detailed thinking {thinking}', 'role': 'system'}] + x['messages']})
```

Ruff (0.11.9)

Single quotes found but double quotes preferred

Replace single quotes with double quotes

(Q000)